feature-libs/requested-delivery-date/root/components/delivery-mode-date-picker/delivery-mode-date-picker.component.ts

```diff
@@ -0,0 +1,156 @@
+/*
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { HttpErrorResponse } from '@angular/common/http';
+import { Component, OnDestroy, OnInit, Optional } from '@angular/core';
+import { FormControl, FormGroup } from '@angular/forms';
+import { Cart } from '@spartacus/cart/base/root';
+import { CheckoutSupportedDeliveryModesQueryReloadEvent } from '@spartacus/checkout/base/root';
+import {
+  CxDatePipe,
+  ErrorModel,
+  EventService,
+  GlobalMessageService,
+  GlobalMessageType,
+  TranslationService,
+} from '@spartacus/core';
+import { Card, OutletContextData } from '@spartacus/storefront';
+import { Observable, Subscription } from 'rxjs';
+import { filter, map } from 'rxjs/operators';
+import { RequestedDeliveryDateFacade } from '../../facade/requested-delivery-date.facade';
+
+@Component({
+  selector: 'cx-request-delivery-date',
+  templateUrl: './delivery-mode-date-picker.component.html',
+  providers: [CxDatePipe],
+})
+export class DeliveryModeDatePickerComponent implements OnInit, OnDestroy {
+  constructor(
+    protected datePipe: CxDatePipe,
+    protected requestedDelDateFacade: RequestedDeliveryDateFacade,
+    protected eventService: EventService,
+    protected translation: TranslationService,
+    protected globalMessageService: GlobalMessageService,
+    @Optional() protected deliveryOutlet?: OutletContextData
+  ) {}
+
+  protected cartEntry: Cart = {};
+  protected subscription = new Subscription();
+
+  protected earliestRetrievalAt: string | undefined;
+  protected requestedRetrievalAt: string | undefined;
+  protected form: FormGroup = new FormGroup({
+    requestDeliveryDate: new FormControl(),
+  });
+  protected isDatePickerReadOnly: boolean = true;
+
+  ngOnInit(): void {
+    if (this.deliveryOutlet?.context$) {
+      this.subscription.add(
+        this.deliveryOutlet.context$.subscribe((context) => {
+          this.cartEntry = context?.item;
+          this.isDatePickerReadOnly = context?.readonly || false;
+        })
+      );
+    }
+
+    if (this.isEarliestRetrievalDatePresent()) {
+      this.earliestRetrievalAt = this.cartEntry.earliestRetrievalAt;
+    }
+    if (this.isRequestedDeliveryDatePresent()) {
+      this.requestedRetrievalAt = this.cartEntry.requestedRetrievalAt;
+    } else {
+      //set the value of requestedRetrievalAt as earliestRetrievalAt and update occ.
+      this.requestedRetrievalAt = this.earliestRetrievalAt;
+      this.setRequestedDeliveryDate();
+    }
+    this.form.patchValue({
+      requestDeliveryDate: this.requestedRetrievalAt,
+    });
+  }
+
+  isEarliestRetrievalDatePresent(): boolean {
+    return (
+      this.cartEntry?.earliestRetrievalAt != null &&
+      this.cartEntry?.earliestRetrievalAt !== undefined &&
+      typeof this.cartEntry?.earliestRetrievalAt === 'string'
```

Comment:
Do we need such strict type checking? Maybe would be more beneficial to move this logic to some separate function. By this we will not duplicate a code in the lines: 85-87. Instead we will be able just re-use a function.

Reply:
agreed, will create a method to reuse the logic.

The type check is added just to ensure that the type is not `Date`, not necessary though.